scripts/protocolFilters/aerodrome.ts

```diff
@@ -1,7 +1,48 @@
+import { BlockField, QueryResponse } from '@envio-dev/hypersync-client'
 import { ReferralEvent } from '../types'
+import { getBlock, getHyperSyncClient } from '../utils'
+import {
+  AERODROME_NETWORK_ID,
+  AERODROME_UNIVERSAL_ROUTER_ADDRESS,
+} from '../calculateRevenue/protocols/aerodrome/constants'
 
-// TODO(ENG-194): Check that the user has made at least one transaction on Aerodrome,
-// and all transactions were made after the referral event.
 export async function filter(event: ReferralEvent): Promise<boolean> {
-  return !!event
+  const client = getHyperSyncClient(AERODROME_NETWORK_ID)
+  const query = {
+    transactions: [
+      { to: [AERODROME_UNIVERSAL_ROUTER_ADDRESS], from: [event.userAddress] },
+    ],
+    fieldSelection: {
+      block: [BlockField.Number],
+    },
+    fromBlock: 0,
+  }
+
+  let hasMoreBlocks = true
+  let foundValidTransaction = false
+
+  while (hasMoreBlocks) {
+    const response: QueryResponse = await client.get(query)
+    if (response.nextBlock === query.fromBlock) {
+      hasMoreBlocks = false
+    } else {
+      query.fromBlock = response.nextBlock
+    }
+
+    for (const block of response.data.blocks) {
+      if (block.number) {
+        const blockData = await getBlock(
+          AERODROME_NETWORK_ID,
+          BigInt(block.number),
+        )
+        if (blockData.timestamp < BigInt(event.timestamp)) {
+          return false // disqualify immediately if any tx is too early
+        } else {
+          foundValidTransaction = true
```

Comment:
Are the results from hypersync guaranteed to be ordered? If so, can't we also just bail early and do a `return true` here?

Reply:
Changed in 0251df522c7aae3608bb41a5aa4f251cbca1af38!